contents/docs/self-host/configure/migrate-postgres-clickhouse.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Migrate from Postgres to ClickHouse
+sidebar: Docs
+showTitle: true
+---
+
+> This documentation is in **Beta**. Please provide feedback via the [PostHog Communty Slack workspace](/slack) or [via a GitHub issue](https://github.com/PostHog/posthog.com/issues).
+
+**PostHog backed by Postgres is now deprecated**. We will continue to provide support to Postgres backed installs but we strongly encourage you to migrate to PostHog backed by ClickHouse for improved performance, to receive new features, and continued support in the future.
+
+## Before you begin
+
+TODOs:
+- Is there a minimum version for PostHog + Postgres and a minimum version for PostHog + ClickHouse? For example, does the Replicator plugin have a minimum PostHog version to work within?
+
+In order to migrate you must have the following in place:
+
+1. The PostHog + Postgres installation
+2. A new PostHog + ClickHouse installation
+
+## Export
+
+TODOs:
+- Are these the correct tables to dump?
+- Should we be using `--data-only` to only dump the data and not the schema creation?
+
+The first step in the migration process is to create a Postgres database dump. With a PostHog + Postgres instance up and running, run the `pg_dump` command with the following format:
+
+```
+$ pg_dump -d [database] -h [host] -U [user] -T posthog_event -T posthog_person -T posthog_sessionrecordingevent -f [output-file]
+```
+
+For example:
+
+```
+$ pg_dump -d posthog -h localhost -U posthog -T posthog_event -T posthog_person -T posthog_sessionrecordingevent -f posthog-migration.sql
+```
+
+The example command above creates a `posthog-migration.sql` that contains SQL statements to create the schema and insert the data data, excluding event data, from the existing Postgres database.
+
+## Import
+
+The next step is to get the data into the new PostHog + ClickHouse installation. This installation still utilizes Postgres, but only for non-event data. Event data is now stored in ClickHouse.
+
+The import process has two steps:
+
+1. Run the exported SQL against the Postgres database for the new installation
+2. Use the PostHog replicator plugin to migrate event data from the old PostHog instance to the new one
+
+### Postgres import of non-event data
+
+Import the old Postgres data into the new PostHog Postgres database. The command should take the following format:
+
+```
+pg_restore -d [database] -h [host] -p [port] -U [user] [input-file]
+```
+
+For example:
+
+```
+pg_restore -d posthog -h remote.example.com -p 5432 -U posthog posthog-migration.sql
+```
+
+### ClickHouse import of event data
+
+TODOs:
+- Does this need to be done for each project or does it handle multiple projects?
```

Comment:
@yakkomajuri - need your input here.

Also, I'm wondering if there is a minimum PostHog version requirement for some of the functionality we want here?

Reply:
Maybe I'm a bit confused here about where the replicator should run?

Does it

1. push the data from the old instance to the new
2. pull from the old instance to the new

?

If it pushes from the old instance to the new then isn't there going to be a problem in that the Replicator plugin isn't going to be compatible with an old instance of PostHog?